runtime/error.c

```diff
@@ -1,9 +1,116 @@
+#include <string.h>
+#include <stdlib.h>
+#include <stdio.h>
+
 #include "balrt.h"
+#include "third-party/libbacktrace/backtrace.h"
+
+#define MIN_FRAME_COUNT 2
+#define SKIP_FROM_END 1 // This number counts from the last stack frame
+#define THREAD 0
+
+typedef struct {
+    uint32_t nFrames;
```

Comment:
"nFrames" and "capacityOfFrames" do not make a nice pair of names, because "n" is abbreviated and "capacityOf" is verbose.